sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/SessionTokenMismatchRetryPolicy.java

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation;
+
+import com.azure.cosmos.CosmosException;
+import com.azure.cosmos.implementation.directconnectivity.TimeoutHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class SessionTokenMismatchRetryPolicy extends RetryPolicyWithDiagnostics {
+
+    private final static Logger LOGGER = LoggerFactory.getLogger(SessionTokenMismatchRetryPolicy.class);
+
+    private static final int BACKOFF_MULTIPLIER = 2;
+
+    private final Duration maximumBackoff;
+    private final TimeoutHelper waitTimeTimeoutHelper;
+    private final AtomicInteger retryCount;
+
+    private Duration currentBackoff;
+
+    public SessionTokenMismatchRetryPolicy(int waitTimeInMilliSeconds)
+    {
+        this.waitTimeTimeoutHelper = new TimeoutHelper(Duration.ofMillis(waitTimeInMilliSeconds));
+        this.maximumBackoff = Duration.ofMillis(Configs.getSessionTokenMismatchMaximumBackoffTimeInMs());
+
+        this.retryCount = new AtomicInteger();
+        this.retryCount.set(0);
+        this.currentBackoff = Duration.ofMillis(Configs.getSessionTokenMismatchInitialBackoffTimeInMs());
+    }
+
+    public SessionTokenMismatchRetryPolicy() {
+        this(Configs.getSessionTokenMismatchDefaultWaitTimeInMs());
+    }
+
+    @Override
+    public Mono<ShouldRetryResult> shouldRetry(Exception e) {
+
+        if (!(e instanceof CosmosException)) {
+            return Mono.just(ShouldRetryResult.noRetry());
+        }
+
+        CosmosException cosmosException = (CosmosException)e;
+
+        if (cosmosException.getStatusCode() != HttpConstants.StatusCodes.NOTFOUND ||
+            cosmosException.getSubStatusCode() != HttpConstants.SubStatusCodes.READ_SESSION_NOT_AVAILABLE) {
+
+            LOGGER.debug(
+                "SessionTokenMismatchRetryPolicy not retrying because StatusCode or SubStatusCode not found.");
+
+            return Mono.just(ShouldRetryResult.noRetry());
+        }
+
+        if (this.waitTimeTimeoutHelper.isElapsed()) {
+
+            LOGGER.info(
+                "SessionTokenMismatchRetryPolicy not retrying because it has exceeded " +
+                    "the time limit. Retry count = {}",
+                this.retryCount);
+
+            return Mono.just(ShouldRetryResult.noRetry());
+        }
+
+        Duration effectiveBackoff = Duration.ZERO;
+
+        // Don't penalize first retry with delay
+        if (this.retryCount.getAndIncrement() > 0)
+        {
```

Comment:
code style: in java "{" should be on the same line as if.

Reply:
ACK - Fixed in next iteartion